src/software/ai/hl/stp/tactic/goalie/goalie_fsm.cpp

```diff
@@ -106,10 +107,42 @@ Point GoalieFSM::findGoodChipTarget(
     {
         chip_target = open_areas[0].origin();
     }
-
     return chip_target;
 }
 
+bool GoalieFSM::shouldEvacuateCrease(const Update &event)
+{
+    Rectangle friendly_defense_area = event.common.world.field().friendlyDefenseArea();
+    Ball ball                       = event.common.world.ball();
+
+    // calculate inflated crease obstacle
+    double robot_radius_expansion_amount =
+        ROBOT_MAX_RADIUS_METERS *
+        robot_navigation_obstacle_config.robot_obstacle_inflation_factor();
+    Rectangle inflated_defense_area =
+        friendly_defense_area.expand(robot_radius_expansion_amount);
+
+    bool ball_in_dead_zone =
+        !contains(friendly_defense_area, ball.position()) &&
+        contains(friendly_defense_area.expand(robot_radius_expansion_amount),
+                 ball.position());
+
+    // goalie should only evacuate crease if there are no enemy robots nearby
+    double safe_distance_multiplier = goalie_tactic_config.safe_distance_multiplier();
+    double nearest_enemy_distance_to_ball = distance(
+        event.common.world.enemyTeam().getNearestRobot(ball.position())->position(),
+        ball.position());
```

Comment:
`getNearestRobot` returns an `optional`. We should check that it has a value before accessing it, since there may be no enemy robots on the field